core/zoom_controls.js

```diff
@@ -157,11 +159,18 @@ Blockly.ZoomControls.prototype.position = function() {
       this.left_ -= metrics.flyoutWidth;
     }
   }
-  this.top_ = metrics.viewHeight + metrics.absoluteTop -
-      this.HEIGHT_ - this.bottom_;
+
   if (metrics.toolboxPosition == Blockly.TOOLBOX_AT_BOTTOM) {
-    this.top_ -= metrics.flyoutHeight;
+    this.top_ = this.verticalSpacing_;
+    this.zoomInGroup_.setAttribute('transform', 'translate(0, 34)');
```

Comment:
Where did these constants come from?

Reply:
All of the zoom groups start in the same location:

![zoomoffsets](https://user-images.githubusercontent.com/25440652/50257511-611e1680-03b0-11e9-8ea5-55621ab32ac4.jpg)

So we have to move them to be in the correct locations: (Left example is for bottom-anchored controls, and right example is for top-anchored controls)

![img_20181219_170315](https://user-images.githubusercontent.com/25440652/50257531-72672300-03b0-11e9-9ca3-5e3766f778ba.jpg)

The offsets are based on the previous offsets.
77 old ln 203
43 old ln 260